enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticator.java

```diff
@@ -82,13 +87,27 @@ private void dropAPIKeyDataFromAPIKeyHeader(RequestContext requestContext) throw
 
     private String getAPIKeyFromRequest(RequestContext requestContext) {
         Map<String, String> headers = requestContext.getHeaders();
-        return ConfigHolder.getInstance().getConfig().getApiKeyConfig().getApiKeyInternalHeaders().stream()
+        return getAPIKeyInternalHeaders(requestContext).stream()
```

Comment:
We need to make sure that the `choreo-api-key` should also work.

Reply:
Yes,  `choreo-api-key` is configured using a config. 
```
private List<String> getAPIKeyInternalHeaders(RequestContext requestContext) {
        APIConfig apiConfig = requestContext.getMatchedAPI();
        String configuredApiKeyHeader = apiConfig.getApiKeyHeader();
        List<String> apiKeyInternalHeaders
                = ConfigHolder.getInstance().getConfig().getApiKeyConfig().getApiKeyInternalHeaders();
        if (StringUtils.isNotEmpty(configuredApiKeyHeader)) {
            return Stream.concat(apiKeyInternalHeaders.stream(), Stream.of(configuredApiKeyHeader))
                    .distinct()
                    .collect(Collectors.toList());
        }
        return apiKeyInternalHeaders;
    }
 ```
 Following method returns a list of header names